examples/post-processing/rs-post-processing.cpp

```diff
@@ -0,0 +1,411 @@
+    original and one for filtered */
+    rs2::points depth_points;
+    rs2::points filtered_points;
+
+    // Declare RealSense pipeline, encapsulating the actual device and sensors
+    rs2::pipeline pipe;
+    // Start streaming with default recommended configuration
+    pipe.start();
+
+    /* save the the time of current frame's arrival in last_time
+    curr will hold the next frame's arrival time */
+    long long last_time = std::chrono::high_resolution_clock::now().time_since_epoch().count();
+    long long curr;
+    
+    // we'll use velocity to rotate the pointcloud for a better view of the filters effects
+    float velocity = 0.3;
+
+    // Decalre filters: decimation, temporal and spatial
+    // decimation - reduces depth frame density;
+    // spatial - edge-preserving spatial smoothing
+    // temporal - reduces temporal noise
+    rs2::decimation_filter dec_filter;
+    rs2::spatial_filter spat_filter;
+    rs2::temporal_filter temp_filter;
+
+    // Declare disparity transform from depth to disparity and vice versa
+    rs2::disparity_transform depth_to_disparity;
+    rs2::disparity_transform disparity_to_depth(false);
+
+    filter_ranges ranges;
+    params params;
+
+    // get ranges of all filters
+    get_filter_ranges(ranges, dec_filter, spat_filter, temp_filter);
+
+    // set filters to defaults values
+    set_defaults(params, ranges);
+
+    while (app) // Application still alive?
+    {
+        // render the user interface: sliders, checkboxes
+        render_ui(w, h, params, ranges);
+
+        // Set the filter options as chosen by the user (or by default)
+        set_filter_options(params, dec_filter, spat_filter, temp_filter);
+
+        rs2::frameset data = pipe.wait_for_frames(); // Wait for next set of frames from the camera
+        rs2::frame depth_frame = data.get_depth_frame(); // find depth frame
+        rs2::frame filtered = depth_frame; // declare frame to apply filters on, while preserving the original depth data
+
+        /* Apply filters. The implemented flow of the filters pipeline is in the following order: apply decimation filter,
+        transform the scence into disparity domain, apply spatial filter, apply temporal filter, revert the results back
+        to depth domain (each post processing block is optional and can be applied independantly).
+        */
+        if (params.do_decimate)
+            filtered = dec_filter.proccess(filtered);
+
+        if (params.do_disparity)
+            filtered = depth_to_disparity.proccess(filtered);
+       
+        if (params.do_spatial)
+            filtered = spat_filter.proccess(filtered);
+
+        if (params.do_temporal)
+            filtered = temp_filter.proccess(filtered);  
+        
+        if (params.do_disparity)
+            filtered = disparity_to_depth.proccess(filtered);
+
+        // Generate the pointcloud and texture mappings
+        depth_points = pc.calculate(depth_frame);
```

Comment:
Doing 2 x PointCloud + 2 x Colorize on HD depth, even without the filters seems a bit too much for a demo with GUI. 
In Debug the UI is not responsive. In Release, the UI is not responsive when Decimate.Magnitude = 1. 
Perhaps we need to restructure the code to run post-processing on a separate thread.